src/EventListener/ReadListener.php

```diff
@@ -51,10 +51,33 @@ public function __construct(CollectionDataProviderInterface $collectionDataProvi
      * Calls the data provider and sets the data attribute.
      *
      * @throws NotFoundHttpException
+     *
+     * @deprecated since version 2.5, to be removed in 3.0.
      */
     public function onKernelRequest(GetResponseEvent $event)
     {
-        $request = $event->getRequest();
+        @trigger_error(sprintf('The method %s() is deprecated since 2.5 and will be removed in 3.0.', __METHOD__), E_USER_DEPRECATED);
+
+        $this->handleEvent($event);
+    }
+
+    /**
+     * Calls the data provider and sets the data attribute.
+     *
+     * @throws NotFoundHttpException
+     */
+    public function handleEvent(/*EventInterface */$event)
+    {
+        if ($event instanceof EventInterface) {
+            $request = $event->getContext()['request'];
```

Comment:
This is a hard coupling to Symfony. We should let the user injecting `RequestStack` instead of populating this key.